xray_test.go

```diff
@@ -660,6 +666,53 @@ func TestXrayOfflineDBSyncV3(t *testing.T) {
 	assert.ErrorContains(t, err, "Invalid stream type")
 }
 
+func TestXrayAuditJasSimpleJson(t *testing.T) {
+	output := testXrayAuditJas(t, string(utils.SimpleJson), "jas")
+	verifySimpleJsonJasResults(t, output, 9, 7, 2, 1)
+}
+
+func TestXrayAuditJasNoViolationsSimpleJson(t *testing.T) {
+	output := testXrayAuditJas(t, string(utils.SimpleJson), "npm")
+	verifySimpleJsonScanResults(t, output, 2, 0)
+	verifySimpleJsonJasResults(t, output, 0, 0, 0, 1)
+}
+
+func testXrayAuditJas(t *testing.T, format string, project string) string {
+	initXrayTest(t, commands.GraphScanMinXrayVersion)
+	tempDirPath, createTempDirCallback := coretests.CreateTempDirWithCallbackAndAssert(t)
+	defer createTempDirCallback()
+	projectDir := filepath.Join(filepath.FromSlash(tests.GetTestResourcesPath()), filepath.Join("xray", project))
+	// Copy the multi project from the testdata to a temp dir
+	assert.NoError(t, fileutils.CopyDir(projectDir, tempDirPath, true, nil))
+	// Configure a new server named "default"
+	createJfrogHomeConfig(t, true)
+	defer cleanTestsHomeEnv()
+	baseWd := changeWD(t, tempDirPath)
+	defer func() {
+		assert.NoError(t, os.Chdir(baseWd))
+	}()
+	return xrayCli.WithoutCredentials().RunCliCmdWithOutput(t, "audit", "--format="+format)
+}
+
+func verifySimpleJsonJasResults(t *testing.T, content string, minIacViolations, minSecrets, minApplicable, minNotApplicable int) {
```

Comment:
can we also test the ignore rule?

Reply:
Already tested internally. One of the JSON test files has a suppression, so I verified that we have one less result.